modules/exploits/linux/local/cve_2021_3490_ebpf_alu32_bounds_check_lpe.rb

```diff
@@ -0,0 +1,179 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GreatRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Post::Linux::Kernel
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Linux eBPF ALU32 32-bit Invalid Bounds Tracking LPE',
+        'Description' => %q{
+          Linux kernels from 5.7-rc1 prior to 5.13-rc4, 5.12.4, 5.11.21, and
+          5.10.37 are vulnerable to a bug in the eBPF verifier's verification
+          of ALU32 operations in the scalar32_min_max_and function when performing
+          AND operations, whereby under certain conditions the bounds of a
+          32 bit register would not be properly updated.
+
+          This can be abused by attackers to conduct an out of bounds read
+          and write in the Linux kernel and therefore achieve arbitrary
+          code execution as the root user.
+
+          The target system must be compiled with eBPF support and not have
+          kernel.unprivileged_bpf_disabled set to 1, which prevents unprivileged
+          users from loading eBPF programs into the kernel. Note that if
```

Comment:
This is no longer true since the introduction of `CONFIG_BPF_UNPRIV_DEFAULT_OFF`. See #15582.

Suggest simply:

```suggestion
          The target system must be compiled with eBPF support and not have
          kernel.unprivileged_bpf_disabled set, which prevents unprivileged
          users from loading eBPF programs into the kernel. Note that if
```

Reply:
Nice find, yeah this should be updated :)

Reply:
Resolved in https://github.com/rapid7/metasploit-framework/pull/15567/commits/db5ac81ecf76e1929fe1ed01a8951b2e00dde97c